drivers/adc/ad7746/ad7746.c

```diff
@@ -504,5 +509,49 @@ int32_t ad7746_get_cap_data(struct ad7746_dev *dev, uint32_t *cap_data)
 		    ((uint32_t)dev->buf[1] << 8) |
 		    dev->buf[0];
 
+	if (dev->setup.config.md == AD7746_MODE_SINGLE)
+		dev->setup.config.md = AD7746_MODE_IDLE;
+
 	return SUCCESS;
 }
+
+/***************************************************************************//**
+ * @brief Perform offset/gain calibration
+ *
+ * @param dev - Device descriptor pointer.
+ * @param md - AD7746 calibration mode specifier.
+ *
+ * @return return code.
+ *         Example: -EINVAL - Wrong input values.
+ *                  -EIO - I2C Communication error.
+ *                  SUCCESS - No errors encountered.
+*******************************************************************************/
+int32_t ad7746_calibrate(struct ad7746_dev *dev, enum ad7746_md md)
+{
+	struct ad7746_config *c = &dev->setup.config;
+	int32_t ret, timeout = 10;
+	uint8_t reg;
+
+	if (md != AD7746_MODE_OFFSET_CALIB &&
+	    md != AD7746_MODE_GAIN_CALIB)
+		return -EINVAL;
+
+	c->md = md;
+	ret = ad7746_set_config(dev, *c);
+	if (ret < 0)
+		return ret;
+
+	do {
+		mdelay(20);
```

Comment:
is this a datasheet related value? or trial/error value? might fit a code comment here :)

Reply:
yeah it needs a comment !